frontend/src/app/core/services/match.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { Match } from 'src/app/shared/models/match.model';
+import data from '../../../assets/mock-data.json';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MatchService {
+
+  constructor() { }
+
+  getSchedule(seasonID: number): Map<number, Match[]> {
+    const schedule: Map<number, Match[]> = new Map<number, Match[]>();
+
+    // Get each match of the given season, grouped in order of what round they take place in
+    const sortedSchedule: Match[][] = [];
+    for (const match of data.matches) {
+      if (match.seasonID === seasonID) {
+        let isNewRound = true;
```

Comment:
A little confused by isNewRound value. If you could post an explanation in the dev chat, I would appreciate it. Or just add a comment.